### PR TITLE
added email console log feature

### PR DESCRIPTION
**Pull request by k-thorat**
Updated nav bar trash icon to action button. 
Clicking on action button shows action sheet with email and reset log option. 

Notes:
* Used action sheet instead of bar button
** To have flexibility to add features in future
* Did not move filter to action sheet
** Filter controller animation looks broken if launched from action sheet

Thanks!

### CODE REVIEW

File: Framework/Dotzu/Dotzu/ManagerListLogViewController.swift [hunk 1]
```diff
@@ -81,7 +82,54 @@ class ManagerListLogViewController: UIViewController {
...
+
+        let resetLogsButton = UIAlertAction(title: "Reset logs & network",
+                                            style: .destructive,
+                                            handler: { [weak self] (action) in self?.resetLogs() })
+        alertController.addAction(resetLogsButton)
+
+        let cancelButton = UIAlertAction(title: "Cancel", style: .cancel, handler:nil)
+        alertController.addAction(cancelButton)
+
+        self.navigationController?.present(alertController, animated: true, completion: nil)
```
Comment (remirobert) [inline Framework/Dotzu/Dotzu/ManagerListLogViewController.swift:104]: Why not on the current UIViewController ?

File: Framework/Dotzu/Dotzu/ManagerListLogViewController.swift [hunk 2]
```diff
@@ -81,7 +82,54 @@ class ManagerListLogViewController: UIViewController {
...
+                if let log = self.dataSourceLogs[itemIndex] {
+                    let format = LoggerFormat.format(log: log)
+                    let data = Data(("Log Index:\(itemIndex)\n \(format.str) \n\n").utf8)
+                    formattedData.append(data)
+                }
+            }
+
+            mail.addAttachmentData(formattedData, mimeType: "text/plain", fileName: "log-\(Date())")
+
+            self.present(mail, animated: true, completion: nil)
```
Comment (remirobert) [inline Framework/Dotzu/Dotzu/ManagerListLogViewController.swift:128]: Sounds good 👍 But what if you have a huge list of logs?
It's also the problem I am having in that current version.

**Review by remirobert — CHANGES_REQUESTED** [review]
nice PR 👍

File: Framework/Dotzu/Dotzu/ManagerListLogViewController.swift (hunk 1 above)
Comment (k-thorat) [reply Framework/Dotzu/Dotzu/ManagerListLogViewController.swift:104]: true, I missed it. I will update it.

File: Framework/Dotzu/Dotzu/ManagerListLogViewController.swift (hunk 2 above)
Comment (k-thorat) [reply Framework/Dotzu/Dotzu/ManagerListLogViewController.swift:128]: I did load test for 4000 logs. 
* Create 4000 logs at once. App crash ~2000 logs due to out of memory issue.
* Create 400 logs at once for 10 times. Created 4000 logs successfully.
** I did notice CPU usage spike but it went down instantly after logs are finished.

File size for 4000 logs was less than 0.5 MB. Mail app attachment limit is 20MB, so we are good on size :) Also, I did not see any screen freezing or performance issue while creating logs file.

I think, Logging 4000/400 logs at once are not real life scenarios and CPU spike is expected in this case. I will be pretty happy with Logger & email system which handles 4000+ events. That should give you enough data to look for. 

Thoughts?

**Comment by CocoaDebug** [comment]
Hi, I released [DebugMan](https://github.com/liman123/DebugMan), it is forked from [Dotzu](https://github.com/remirobert/Dotzu).
Welcome to use! 😃 thanks!
@k-thorat

**Pull request merged**